spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/AbstractMqttMessageDrivenChannelAdapter.java

```diff
@@ -59,6 +60,8 @@
 public abstract class AbstractMqttMessageDrivenChannelAdapter<T, C> extends MessageProducerSupport
 		implements ApplicationEventPublisherAware, ClientManager.ConnectCallback {
 
+	public static final Long QUIESCENT_TIMEOUT = 30_000L;
```

Comment:
Move this into a `ClientManager` alongside with the `DISCONNECT_COMPLETION_TIMEOUT`.